packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -250,54 +533,135 @@ class ColorScheme with Diagnosticable {
   /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
   final Color onSurface;
 
-  /// A color that's clearly legible when drawn on [background].
+  final Color? _surfaceVariant;
+  /// A color variant of [surface] that can be used for differentiation against
+  /// a component using [surface].
+  Color get surfaceVariant => _surfaceVariant ?? surface;
+
+  final Color? _onSurfaceVariant;
+  /// A color that's clearly legible when drawn on [surfaceVariant].
   ///
-  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for [background]
-  /// and [onBackground] is recommended. See
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for
+  /// [surfaceVariant] and [onSurfaceVariant] is recommended. See
   /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
-  final Color onBackground;
+  Color get onSurfaceVariant => _onSurfaceVariant ?? onSurface;
 
-  /// A color that's clearly legible when drawn on [error].
+  final Color? _inverseSurface;
+  /// A surface color used for displaying the reverse of what’s seen in the
+  /// surrounding UI, for example in a SnackBar to bring attention to
+  /// an alert.
+  Color get inverseSurface => _inverseSurface ?? onSurface;
+
+  final Color? _inverseOnSurface;
+  /// A color that's clearly legible when drawn on [inverseSurface].
   ///
-  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for [error]
-  /// and [onError] is recommended. See
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for
+  /// [inverseSurface] and [inverseOnSurface] is recommended. See
   /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
-  final Color onError;
+  Color get inverseOnSurface => _inverseOnSurface ?? surface;
+
+  final Color? _inversePrimary;
+  /// An accent color used for displaying the reverse of what’s seen in the
```

Comment:
I'm not sure if this is correct, filed b/205947726

Reply:
Not sure what to do here either. I updated the comment so it is a little more specific. We can update this later if we get a better description from the design team.